Tweak auth warning. (#323)
cylc/cylc-doc: src/user-guide/sharing-access-to-workflows.rst

The site_authorization configuration allows sites to configure sensible defaults
and limits for the permissions users can delegate.

> Note that as the UI Server runs as the workflow owner, s/he has full control

Reviewer: ```suggestion
Note that as the UI Server runs as the workflow owner, they have full control
```
